src/dbally/prompts/few_shot.py

```diff
@@ -0,0 +1,37 @@
+import inspect
+import re
+import textwrap
+from typing import Callable, Union
+
+
+class FewShotExample:
+    """
+    A question:answer representation for few-shot prompting
+    """
+
+    def __init__(self, question: str, answer_expr: Union[str, Callable]) -> None:
+        """
+        Args:
+            question: sample question
+            answer_expr: it can be either a stringified expression or a lambda for greater safety and code completions.
+                in lambda case, it should be a single line expression (otherwise output correctness is not guaranteed)
+        """
+        self.question = question
+        self.answer_expr = answer_expr
+
+        if isinstance(self.answer_expr, str):
+            self.answer = self.answer_expr
+        else:
+            expr_source = textwrap.dedent(inspect.getsource(self.answer_expr))
```

Comment:
This code in the `else:` block effectively cleans up and normalizes a piece of Python code. I think you could extract it into a separate method or at least briefly describe what it does in a doc string.

Reply:
good point, I will fix that in next commit